docs/DEV.md

```diff
@@ -1,33 +1,115 @@
+
+3. **Subgraph Instances**: There is one subgraph instance for each blockchain to efficiently query blockchain data.
+
+4. **IPFS**: IPFS is utilized for reading metadata files, providing decentralized file storage.
+
+5. **[Pinata](https://www.pinata.cloud/)**: Pinata is used to upload and pin files to IPFS, ensuring the availability of data.
+
+
+
+Depending on the configuration in your local `.env` file, you can choose to use these services directly or opt for a local version of them.
+
+For faster development and a more responsive feedback loop, we recommend setting up a local development environment instead of relying on testnets
+and external services directly.
+We have provided a Docker Compose configuration that allows you to run local instances of the necessary services.
+
+⚠️ Please note that the local development environment is still a work in progress,
+and not all functionality may be available locally.
+We are continuously improving it to make your development experience as seamless as possible.
+
+
+# Dependencies
+
+* node
+* docker
+
+### Setup your wallet
+
+All the contracts and test data in the local chains are owned by accounts derived from the same test mnemonic phrase.
+To avoid confusion with your real accounts, you can create a new profile in your browser, install Metamask or any other wallets
+that manages [hierarchical-deterministic wallets](https://github.com/bitcoin/bips/blob/master/bip-0032.mediawiki), and import the following test mnemonic to be able to derive all the used accounts:
+
+```
+test test test test test test test test test test test junk
+```
+
+Contracts are deployed with the default account at derivation path `m/44'/60'/0'/0/0`.
+The same accounts owns some ready to use data, like projects, rounds, etc.
+
+⚠️⚠️⚠️
+When you start your local services, the main account's nonce gets reset,
+so you should reset it manually in your wallet to make sure you don't accidentally send transactions with the wrong nonce.
+
+If you use metamask keep your expanded view open:
+
+`Menu -> Expand view`
+
+And reset the wallet every time you restart the local chains:
+
+`Menu -> Settings -> Advance -> CLear activity tab data`
+⚠️⚠️⚠️
+
+
+
+
+
+
+### Run local services
+
+Clone the repo:
 
 ```sh
-$ pnpm install
+git clone git@github.com:gitcoinco/grants-stack.git
+cd grants-stack
 ```
 
+Run the services with `docker-compose` and follow the logs:
+
 ```sh
-cp .env.example .env
+docker-compose up -d && docker-compose logs -f
 ```
 
-The .env file will be prefilled with some static and public variables. For the ones that are empty, please create accounts and fill in your personal API keys for the respective services.
+To kill the containers run:
 
-### Run in Development
+```sh
+docker-compose kill
+```
 
-Runs the app in the development mode.\
-Open [http://localhost:3000](http://localhost:3000) to view it in the browser.
+### Setup the dapps dependencies
```

Comment:
Would probably say "root project dependencies" to distinguish that from the runtime dependencies mentioned above (Pinata, Indexer, etc)